### PR TITLE
HW4: Middleware. Frameworks

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,16 +1,43 @@
...
-const dirWatcher = new DirWatcher();
+const app = express();
 
-dirWatcher.watch('./data/**/*.csv', 10000);
-dirWatcher.on('dirwatcher:changed', (path) => {
-  (async () => {
-    const result = await Importer.import(path);
-    logger.info(`ASYNC ${path} ${result}`);
-  })();
+plainTextServer(8091);
```
Comment (boccob) [inline src/app.js]: I'd move all these servers initialization to index.js. Because they are not connected with rest server.

File: src/app.js [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
 
-  logger.info(`SYNC ${path} ${Importer.importSync(path)}`);
-
-  logger.info(`STREAM ${path}`);
-  Streams.transformFile(path);
+app.use((req, res, next) => {
+  logger.debug(`${req.method} ${req.url}`);
+  next();
 });
+app.use(express.json());
```
Comment (boccob) [inline src/app.js]: Could be a good idea to support also x-www-form-urlencoded

File: src/controllers/productController.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import _ from 'underscore';
```
Comment (boccob) [inline src/controllers/productController.js]: I'd recommend using lodash instead. Or Array methods (find and filter are already there).

File: src/helpers/routeHelper.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import config from '../config/config.json';
+import { ProductController, ReviewController, UserController } from '../controllers';
+
+export class ProductRouteUtils {
+  static addProduct(req, res) {
+    const product = ProductController.addProduct(req.body.name, req.body.brand, req.body.price);
```
Comment (boccob) [inline src/helpers/routeHelper.js:6]: Maybe it's ok but at the moment it's possible to add a product with all empty fields. I'd add some validation.

Comment (ghost) [reply src/helpers/routeHelper.js:6]: Done it

**Review by boccob — APPROVED** [review]

**Pull request merged**